library/idr/attributes.py

```diff
@@ -0,0 +1,185 @@
+"""
+Helper functions for accessing the IDR from within IPython notebooks.
+"""
+from idr import createHTTPsession
+import pandas
+from pandas import DataFrame
+from pandas import read_csv
+from pandas import concat
+
+def attributes_by_attributes(conn,
+                             name="Gene Symbol",
+                             value="ASH2L",
+                             ns="openmicroscopy.org/mapr/gene",
+                             ns2="openmicroscopy.org/mapr/phenotype",
+                             name2=None,
+                             sId=None
+                             ):
+
+    """
+    Return a list of neighbours attributes
+    for given case insensitive attribute value. (Uses the python blitz gateway)
+    """
+    from omero.rtypes import rstring, rlist, unwrap
+    from omero.sys import ParametersI
+
+    params = ParametersI()
+    params.addString("value", value.lower())
+    q = (
+        "select distinct new map( mv.value as value) "
+        "from Annotation as a "
+        "join a.mapValue as mv "
+        "where lower(mv.value) = :value {where_claus}"
+    )
+    where_claus = []
+    if name:
+        params.addString("name", name)
+        where_claus.append("and mv.name = :name")
+    q = q.format(**{'where_claus':" ".join(where_claus)})
```

Comment:
`q.format(where_claus=" ".join(where_claus))` is a slightly cleaner way to write this (and similarly for all the other `format(**{`).

Another alternative (up to you) is to append to the string directly:
```
where_claus = ''
if name:
    params.addString("name", name)
    where_claus += " and mv.name = :name"
```